contracts/lazyclaim/IERC721LazyClaim.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/// @author: manifold.xyz
+
+/**
+ * Lazy Claim interface
+ */
+interface IERC721LazyClaim {
```

Comment:
We should pack the struct!  How many are we realistically going to allow ppl to claim?

Dates can be uint48 (meaning that's 96 bits).  Can we pack totalMax, walletMax into the remaining 160 bits?
Also, put the string at the very end.

Meaning the storage will cost 4 slots:
32byte merkleRoot
32byte for totalMax, walletMax, startDate, endDate
arweave URI

But, even better, can we be SUPER smart about things to minimize things:
arweave hash is 43 characters
So, if we store only the arweave tx hash and super compress, that's even better.

Reply:
Acutally, store the current mint count in here too.

Reply:
In fact, the struct should also have a single byte enum value for 'storageProtocol', enum of
enum StorageProtocol {
  ARWEAVE = 1,
  IPFS = 2
}

And take this as an input.  This way , we can minimize the cost to store the URI.

Reply:
Done. Counters are uint32, total size of the stored struct should be 208 bits (including current total counter)

I also left the zero slot of StorageProtocol for NONE in the event someone wants to use their own server